### ISSUE TITLE
Posible issues with catching errors

### ISSUE THREAD

**Issue by SavaNDragos** [Code-Artist/CodeArtEng.Diagnostics#1]
The implementation of process_ErrorDataReceived will not always catch an error even if it exist.
Example: calling msbuild; msbuild returns everything in the standard output
Using (sender as Process).ExitCode != 0 also doesn't indicate an error all the time because there are some executables like robocopy who can have 0, 1, 2 as succes exit codes.

**Comment by Code-Artist** [comment]
Yes, you are right for both cases. Do you have any recommendation for better implementation?
I will improve this in next release.

**Comment by Code-Artist** [comment]
Hello SavNDragos,

Here are my feedback regarding your concern after review back my ProcessExecutor implementation again.
- process_ErrorDataReceived is solely meant for capture std:err output and ProcessResult.ErrorDetected property will only be set if any std:err output is received. This only applicable for process which make use of std:err to report error. For the case of msbuild, the ErrorDetected flag will never be set.
- Regarding the ExitCode, please take note that ProcessExecutor did not make any decision on ExitCode. It's depends on the host application to define the meaning of exit code. In the case of RoboCopy, you may refer to this [example ](http://ss64.com/nt/robocopy-exit.html) for error handling.

In conclusion, no additional fixes is required for these 2 topics.
Thank you.
